File: TY.Services.Bank/TY.Services.Bank/Models/Transaction/Transaction.cs
using System;

namespace TY.Services.Bank.Models.Transaction
{
    public class Transaction : IBaseEntity
    {
        public string Id { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public Account.Account Account{ get; set; }
        public decimal Amount { get; set; }
        public decimal OldBalance { get; set; }
        public decimal CurrentBalance { get; set; }
        public TransactionTypeEnum TransactionType { get; set; }
    }
}

File: TY.Persistent.Bank/models/Transaction.js
import { set, Schema, model } from 'mongoose'
set('debug', true)

const TransactionSchema = new Schema({
    account: {
        type: Schema.Types.ObjectId,
        ref: 'account'
    },
    amount: {
        type: Number,
        required: true,
        default: 0
    },
    oldBalance: {
        type: Number,
        required: true,
        default: 0
    },
    currentBalance: {
        type: Number,
        required: true,
        default: 0
    },
    transactionType: {
        type: Number,
        required: true,
        min: 0,
        max: 1,
        default: 0
    },
    isActive: {
        type: Boolean,
        default: true
    }
}, {
    timestamps: true,
    toJSON: {
        virtuals: true
    }
});

model('Transaction', TransactionSchema)